ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -696,9 +696,16 @@ private void cleanupExpiredInFlightRequests() {
           correlationIdToChunkPutRequestInfo.entrySet().iterator();
       while (inFlightRequestsIterator.hasNext()) {
         Map.Entry<Integer, ChunkPutRequestInfo> entry = inFlightRequestsIterator.next();
-        if (time.milliseconds() - entry.getValue().startTimeMs > routerConfig.routerRequestTimeoutMs) {
+        if (time.milliseconds() - entry.getValue().getStartTimeMs() > routerConfig.routerRequestTimeoutMs) {
           operationTracker.onResponse(entry.getValue().replicaId, false);
           chunkException = new RouterException("Timed out waiting for responses", RouterErrorCode.OperationTimedOut);
+          ReplicaId replica = entry.getValue().replicaId;
+          DataNodeId dataNodeId = replica.getDataNodeId();
+          NonBlockingRouterMetrics.NodeLevelMetrics dataNodeBasedMetrics =
+              routerMetrics.getDataNodeBasedMetrics(dataNodeId);
+          dataNodeBasedMetrics.putRequestTimeoutCount.inc();
```

Comment:
these node level metrics are kinda distracting. let us try to limit these and keep them lean.